controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -118,9 +118,31 @@ else if (retryBecauseNewFailure(application, report.jobType()))
     /** Returns whether all production zones listed in deployment spec last were successful on the currently deploying change. */
     private boolean deploymentComplete(LockedApplication application) {
         if ( ! application.deploying().isPresent()) return true;
```

Comment:
Isn't the doc in disagreement with this return? When an upgrade is cancelled, it wasn't _successful on the ... change_, but it does clear the `deploying()`.

Reply:
In that case the deployment is incomplete and this method returns false - so how is it in disagreement?

Reply:
No, it returns `true`.

Reply:
No but perhaps you mean if it was cancelled and then later we check for another Change which is an upgrade to an earlier version? Then it will return true but then it is also true that the upgrade is complete ...

Reply:
The doc is good enough now, after you changed it.